c_glib/test/test-memory-mapped-file.rb

```diff
@@ -22,9 +22,8 @@ def test_open
     tempfile.close
     file = Arrow::MemoryMappedFile.open(tempfile.path, :read)
     begin
-      buffer = " " * 5
-      file.read(buffer)
-      assert_equal("Hello", buffer)
+      buffer = file.read(5)
+      assert_equal("Hello", buffer.data.to_s)
```

Comment:
You talk about removed code not added code, don't you?

It was implemented by just passing `char *` internal buffer of Ruby's `String` as a buffer space to [`garrow_readable_read()'s buffer`](https://github.com/apache/arrow/pull/618/files/e14bb40bf3f34abb261ebd4f7a3b4e69f486cf14#diff-3c4862f397803c3c5de06fd7f1c5a672L63). We sometime uses `String` as a buffer in Ruby. For example, [the `IO#read`'s `outbuf` argument](https://docs.ruby-lang.org/en/2.4.0/IO.html#method-i-read).

It's safe usage but we need to allocate the `char *` internal buffer before we use it. It's not easy to use for users. :<